test/dlrm/lds_mapping_util.hpp

```diff
@@ -193,34 +193,34 @@ namespace rocwmma
 
         __device__ static inline uint32_t baseOffsetB()
         {
-            auto matrixCoord = std::make_pair(
-                LocalWriteFragA::size() * BlocksX * std::get<0>(LdsOffsets::workgroupDim()), 0);
```

Comment:
- From a design standpoint, the std::get<Obj, N> is a better generic interface for element access as opposed to named locals.

Why? Straight usage of coord.x for example ensures that coordinates used in this context MUST have a public local variable x. If we change the coord class object (such as what we're doing now), then we need to find ALL the instances of .x and change it to whatever other member is the replacement. This is called tight coupling.

If we continue with the std::get<Obj, N> interface, we just need a specialization for the new Obj type to correctly map to the new location of x.

```
// Contextual access
enum : uint32_t 
{
    X_COORD_INDEX = 0,
    Y_COORD_INDEX = 1
};

template<>
class get<OldObj, X_COORD_INDEX > (OldObj& o) { return o.x; }
template<>
auto& get<OldObj, Y_COORD_INDEX > (OldObj& o) { return o.y; }

template<>
auto& class get<NewObj, X_COORD_INDEX > (NewObj& o) { return o.getXMember(); }

template<>
auto& class get<NewObj, Y_COORD_INDEX > (NewObj& o) { return o.getYMember(); }
```

Now, regardless of the coord object you choose to use, std::get<X_COORD_INDEX>(obj) consolidates contextual x coordinate access behind the same interface. This is a very important concept for object oriented design.

That being said: we are eyeing the removal of std:: library dependence due to hipRTC.
This removal dictates that we can't use std headers, but we can quickly come up with a drop-in replacement for get with the same interface.